dbgd.c

```diff
@@ -250,3 +251,27 @@ static int dbgd_show_front_screen(Dbg__Request *req, Dbg__Response *res)
 
     return DBG__RESPONSE__TYPE__OK;
 }
+
+static int dbgd_scsi_dvd_in(Dbg__Request *req, Dbg__Response *res)
+{
+    if (!req->has_command || !req->has_size)
+        return DBG__RESPONSE__TYPE__ERROR_INCOMPLETE_REQUEST;
+
+    res->buffer.len  = req->size; 
+    res->buffer.data = malloc(res->buffer.len);
+    res->has_buffer = 1;
+    NTSTATUS status = scsi_dvd_command(SCSI_IOCTL_DATA_IN, req->command.data, req->command.len, res->buffer.data, res->buffer.len);
```

Comment:
Have you looked at the disassembly for `scsi_dvd_command`? My guess is that it's a wrapper around some IO port stuff. It would be cool to just implement IO port read/write and do the SCSI stuff dev box side.